### PR TITLE
Accelerate import updates for iOS

### PR DESCRIPTION
**Pull request by jagrit06**
## Proposed changes

Fixes issues from #1217

## Checklist

Put an `x` in the boxes that apply.

- [x] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [x] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation (if needed)

### CODE REVIEW

**Review by awni — APPROVED** [review]
Looks great, thanks for fixing that! @davidkoski can you check that it compiles now with your MLX Swift branch?

**Comment by davidkoski** [comment]
Yes, will do

**Comment by davidkoski** [comment]
I still get the error -- it looks like the `float16_t` is still resolving to `_MLX_Float16`

**Review by awni — CHANGES_REQUESTED** [review]
Could you update this to use https://github.com/ml-explore/mlx/blob/main/mlx/backend/common/softmax.cpp#L110 so that we have the same numerical behavior across platforms?

File: mlx/backend/accelerate/softmax.cpp [hunk 1]
```diff
@@ -53,25 +56,26 @@ inline simd_float16 simd_fast_exp(simd_float16 x) {
...
 /**
  * The ARM neon equivalent of the fast exp above.
  */
 inline float16x8_t neon_fast_exp(float16x8_t x) {
-  x = vmulq_f16(x, vdupq_n_f16(1.442695)); // multiply with log_2(e)
-  x = vmaxq_f16(x, vdupq_n_f16(-14)); // clamp under with -14
-  x = vminq_f16(x, vdupq_n_f16(14)); // clamp over with 14
+  x = vmulq_f16(x, vdupq_n_f16(float16_t(1.442695f))); // multiply with log_2(e)
+  x = vmaxq_f16(x, vdupq_n_f16(float16_t(-14.f))); // clamp under with -14
+  x = vminq_f16(x, vdupq_n_f16(float16_t(14.f))); // clamp over with 14
```
Comment (davidkoski) [inline mlx/backend/accelerate/softmax.cpp:66]: Maybe these changes should be reverted?  They don't fix it and maybe just make it less clear?

Comment (jagrit06) [reply mlx/backend/accelerate/softmax.cpp:66]: So they do fix things on some versions of iOS, so I will leave them there to be safe

**Review by awni — APPROVED** [review]
Thanks for the fix!!

**Pull request merged**